controllers/topology/builtin.go

```diff
@@ -297,7 +297,7 @@ func (s *BuiltInTopologyService) EditTopology(replicas map[string]*corev1.Pod, r
 
 			roles, err = GetRoles(pod)
 			if err != nil {
-				return false, err
+				return Replicaset{}, err
```

Comment:
It is more efficient to return a reference from a function and a nil value in case of an error